libcontainerd/process_windows.go

```diff
@@ -63,26 +65,24 @@ func (w *delToBsWriter) Write(b []byte) (int, error) {
 	return w.WriteCloser.Write(bc)
 }
 
-type stdInCloser struct {
-	io.WriteCloser
-	hcsshim.Process
-}
-
-func createStdInCloser(pipe io.WriteCloser, process hcsshim.Process) *stdInCloser {
-	return &stdInCloser{
-		WriteCloser: pipe,
-		Process:     process,
-	}
-}
-
-func (stdin *stdInCloser) Close() error {
-	if err := stdin.WriteCloser.Close(); err != nil {
-		return err
-	}
+func createStdInCloser(pipe io.WriteCloser, process hcsshim.Process) io.WriteCloser {
+	return ioutils.NewWriteCloserWrapper(pipe, func() error {
+		if err := pipe.Close(); err != nil {
+			return err
+		}
 
-	return stdin.Process.CloseStdin()
-}
+		// We do not need to lock container ID here, even though
+		// we are calling into hcsshim. This is safe, because the
+		// only place that closes this process handle is this method.
+		err := process.CloseStdin()
+		if err != nil && !hcsshim.IsNotExist(err) {
+			// This error will occur if the compute system is currently shutting down
+			if perr, ok := err.(*hcsshim.ProcessError); ok && perr.Err != hcsshim.ErrVmcomputeOperationInvalidState {
+				logrus.Error(err)
```

Comment:
The caller (`AttachStreams` in monitor.go) ignores errors returned from calls to `Close()`, but the errors found by logging them have proven valuable to finding bugs in this area. I'd like to have this logged somewhere. I can either change this to a Debug rather than Error, or all `Close()` errors in AttachStreams. WDYT?